### PR TITLE
args: add `init_bulk_jdata_arginfo` and `init_surf_jdata_arginfo`

### PR DESCRIPTION
**Pull request by HuangJiameng**
Arginfo and two examples on `init` have been updated. But there are still a few problems.
1. How does parameter "out_dir", "type_map", "atom_masses" work in init process?
2. Which kind of example does `al.json` belong to? Should "stages" be added into it?
3. I am unfamiliar with parameter "mid_point", "lmp_cmd" in cu.surf.hcp.111.json, please add it into arginfo.
4. Is parameter "super_cell" necessary when `from_poscar` is True?
5. Explanations for "head_ratio" and "vacuum_numb"?

### CODE REVIEW

File: dpgen/data/arginfo.py [hunk 1]
```diff
@@ -35,6 +35,119 @@ def init_reaction_mdata_arginfo() -> Argument:
...
+        Argument("md_incar", str, optional=True, doc=doc_md_incar),
+        Argument("scale", list, optional=False, doc=doc_scale),
+        Argument("skip_relax", bool, optional=False, doc=doc_skip_relax),
+        Argument("pert_numb", int, optional=False, doc=doc_pert_numb),
+        Argument("pert_box", float, optional=False, doc=doc_pert_box),
+        Argument("pert_atom", float, optional=False, doc=doc_pert_atom),
+        Argument("md_nstep", int, optional=False, doc=doc_md_nstep),
+        Argument("coll_ndata", int, optional=False, doc=doc_coll_ndata),
+        Argument("type_map", list, optional=True, doc=doc_type_map),
+        Argument("init_fp_style", str, optional=True, default="VASP", doc=doc_init_fp_style),
```
Comment (njzjz) [inline dpgen/data/arginfo.py:86]: Use [`Variant`](https://docs.deepmodeling.com/projects/dargs/en/stable/api/dargs.html#dargs.Variant) instead, and put `relax_kpt`, `md_kpt`, and `atom_masses` into the `Argument` of ABACUS style.

**Comment by njzjz** [comment]
I am not familiar with these modules as well. You may request others' reviews

**Comment by njzjz** [comment]
Also, they should be added into the documentation.

File: dpgen/data/arginfo.py [hunk 2]
```diff
@@ -35,6 +35,119 @@ def init_reaction_mdata_arginfo() -> Argument:
...
+    doc_scale = "Scales for transforming cells."
+    doc_skip_relax = "If it's true, you may directly run stage 2 (perturb and scale) using an unrelaxed POSCAR."
+    doc_pert_numb = "Number of perturbations for each POSCAR."
+    doc_pert_box = "Percentage of Perturbation for cells."
+    doc_pert_atom = "Perturbation of each atoms (Angstrom)."
+    doc_md_nstep = "Steps of AIMD in stage 3. If it's not equal to settings via `NSW` in `md_incar`, DP-GEN will follow `NSW`."
+    doc_coll_ndata = "Maximal number of collected data."
+    doc_type_map = "The indices of elements in deepmd formats will be set in this order."
+    doc_init_fp_style = "First-principle software. If this key is absent."
+    doc_relax_kpt = 'Path of `KPT` file for relaxation in stage 1. Only useful if `init_fp_style` is "ABACUS".'
```
Comment (hongriTianqi) [inline dpgen/data/arginfo.py:65]: `relax_kpt` has been deleted.

File: dpgen/data/arginfo.py [hunk 3]
```diff
@@ -35,6 +35,119 @@ def init_reaction_mdata_arginfo() -> Argument:
...
+    doc_md_kpt = 'Path of `KPT` file for MD simulations in stage 3. Only useful if `init_fp_style` is "ABACUS".'
```
Comment (hongriTianqi) [inline dpgen/data/arginfo.py:66]: `md_kpt` has been deleted.

**Comment by hongriTianqi** [comment]
> Arginfo and two examples on `init` have been updated. But there are still a few problems.
> 
> 1. How does parameter "out_dir", "type_map", "atom_masses" work in init process?

As with "atom_masses", ABACUS need to be fed with an atomic mass list when generating supercells, because the `STRU` file needs element's mass.

File: dpgen/data/arginfo.py [hunk 4]
```diff
@@ -35,6 +35,135 @@ def init_reaction_mdata_arginfo() -> Argument:
...
+    doc_super_cell = "Size of supercell."
+    doc_from_poscar = "Deciding whether to use a given poscar as the beginning of relaxation. If it's true, keys (`cell_type`, `latt`) will be aborted. Otherwise, these two keys are **necessary**."
+    doc_from_poscar_path = "Path of POSCAR for VASP or STRU for ABACUS. **Necessary** if `from_poscar` is true."
+    doc_latt = "Lattice constant for single cell."
+    doc_layer_numb = "Number of equivalent layers of slab."
+    doc_z_min = "Thickness of slab without vacuum (Angstrom). If the layer_numb and z_min are all set, the z_min value will be ignored."
+    doc_vacuum_max = "Maximal thickness of vacuum (Angstrom)."
+    doc_vacuum_min = "Minimal thickness of vacuum (Angstrom). Default value is 2 times atomic radius."
+    doc_vacuum_resol = "Interval of thickness of vacuum. If size of `vacuum_resol` is 1, the interval is fixed to its value. If size of `vacuum_resol` is 2, the interval is `vacuum_resol[0]` before `mid_point`, otherwise `vacuum_resol[1]` after `mid_point`."
+    doc_vacuum_numb = "Number of vacuum. **Necessary** if vacuum_resol is empty."
```
Comment (wanghan-iapcm) [inline dpgen/data/arginfo.py:130]: What is "number of vacuum"? please explain.

File: dpgen/data/arginfo.py [hunk 5]
```diff
@@ -35,6 +35,135 @@ def init_reaction_mdata_arginfo() -> Argument:
...
+    doc_stages = "Stages for `init_surf`."
+    doc_elements = "Atom types."
+    doc_potcars = "Path of POTCAR."
+    doc_cell_type = "Specifying which typical structure to be generated. **Options** include fcc, hcp, bcc, sc, diamond."
+    doc_super_cell = "Size of supercell."
+    doc_from_poscar = "Deciding whether to use a given poscar as the beginning of relaxation. If it's true, keys (`cell_type`, `latt`) will be aborted. Otherwise, these two keys are **necessary**."
+    doc_from_poscar_path = "Path of POSCAR for VASP or STRU for ABACUS. **Necessary** if `from_poscar` is true."
+    doc_latt = "Lattice constant for single cell."
+    doc_layer_numb = "Number of equivalent layers of slab."
+    doc_z_min = "Thickness of slab without vacuum (Angstrom). If the layer_numb and z_min are all set, the z_min value will be ignored."
```
Comment (wanghan-iapcm) [inline dpgen/data/arginfo.py:126]: What is "Thickness of slab without vacuum" ? Do you mean the distance between the top layer and bottom layer atoms? 

"If the layer_numb and z_min are all set, the z_min value will be ignored."
Should be a typo

File: dpgen/data/arginfo.py [hunk 6]
```diff
@@ -35,6 +35,135 @@ def init_reaction_mdata_arginfo() -> Argument:
...
+    doc_vacuum_max = "Maximal thickness of vacuum (Angstrom)."
+    doc_vacuum_min = "Minimal thickness of vacuum (Angstrom). Default value is 2 times atomic radius."
+    doc_vacuum_resol = "Interval of thickness of vacuum. If size of `vacuum_resol` is 1, the interval is fixed to its value. If size of `vacuum_resol` is 2, the interval is `vacuum_resol[0]` before `mid_point`, otherwise `vacuum_resol[1]` after `mid_point`."
+    doc_vacuum_numb = "Number of vacuum. **Necessary** if vacuum_resol is empty."
+    doc_mid_point = "Value of mid-point. **Necessary** if the size of vacuum_resol is 2 or 0."
+    doc_head_ratio  = "Ratio of head. **Necessary** if vacuum_resol is empty."
```
Comment (wanghan-iapcm) [inline dpgen/data/arginfo.py:132]: please explain what is "head"

File: dpgen/data/arginfo.py [hunk 7]
```diff
@@ -35,6 +35,135 @@ def init_reaction_mdata_arginfo() -> Argument:
...
+    """
+    doc_init_surf = "Generate initial data for surface systems."
+    doc_stages = "Stages for `init_surf`."
+    doc_elements = "Atom types."
+    doc_potcars = "Path of POTCAR."
+    doc_cell_type = "Specifying which typical structure to be generated. **Options** include fcc, hcp, bcc, sc, diamond."
+    doc_super_cell = "Size of supercell."
+    doc_from_poscar = "Deciding whether to use a given poscar as the beginning of relaxation. If it's true, keys (`cell_type`, `latt`) will be aborted. Otherwise, these two keys are **necessary**."
+    doc_from_poscar_path = "Path of POSCAR for VASP or STRU for ABACUS. **Necessary** if `from_poscar` is true."
+    doc_latt = "Lattice constant for single cell."
```
Comment (wanghan-iapcm) [inline dpgen/data/arginfo.py:124]: I think we should make the `cell_type` and `latt` optional, probably in a separate PR.

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]

File: dpgen/data/arginfo.py (hunk 7 above)
Comment (HuangJiameng) [reply dpgen/data/arginfo.py:125]: github.com/deepmodeling/dpgen/pull/930 `cell_type` is optional now.

**Review by njzjz — COMMENTED** [review]
Can you add the parameter document?

**Review by wanghan-iapcm — APPROVED** [review]

**Comment by njzjz** [comment]
> Can you add the parameter document?

Do you need help?

**Review by hongriTianqi — APPROVED** [review]

**Comment by HuangJiameng** [comment]
> > Can you add the parameter document?
> 
> Do you need help?

Sorry for missing the message. Isn't parameter document added in `doc_<param-name>` in arginfo?

**Comment by njzjz** [comment]
Refer `doc/init/init-bulk-mdata.rst`.

**Comment by njzjz** [comment]
Adding TOC?

**Review by njzjz — APPROVED** [review]

**Review by Liu-RX — APPROVED** [review]

File: dpgen/data/arginfo.py [hunk 8]
```diff
@@ -35,6 +36,135 @@ def init_reaction_mdata_arginfo() -> Argument:
...
+    doc_layer_numb = "Number of equivalent layers of slab."
+    doc_z_min = "Thickness of slab without vacuum (Angstrom). If `layer_numb` is set, `z_min` will be ignored."
+    doc_vacuum_max = "Maximal thickness of vacuum (Angstrom)."
+    doc_vacuum_min = "Minimal thickness of vacuum (Angstrom). Default value is 2 times atomic radius."
+    doc_vacuum_resol = "Interval of thickness of vacuum. If size of `vacuum_resol` is 1, the interval is fixed to its value. If size of `vacuum_resol` is 2, the interval is `vacuum_resol[0]` before `mid_point`, otherwise `vacuum_resol[1]` after `mid_point`."
+    doc_vacuum_numb = "The total number of vacuum layers **Necessary** if vacuum_resol is empty."
+    doc_mid_point = "The mid point of head region and tail region. **Necessary** if the size of vacuum_resol is 2 or 0."
```
Comment (Vibsteamer) [inline dpgen/data/arginfo.py:132]: The mid point of 
->
The mid point separating

File: dpgen/data/arginfo.py [hunk 9]
```diff
@@ -35,6 +36,135 @@ def init_reaction_mdata_arginfo() -> Argument:
...
+    doc_head_ratio  = "Ratio of vacuum layers in the nearby region with denser intervals(head region). **Necessary** if vacuum_resol is empty."
+    doc_millers = "Miller indices."
+    doc_relax_incar = "Path of INCAR for relaxation in VASP. **Necessary** if `stages` include 1."
+    doc_scale = "Scales for transforming cells."
```
Comment (Vibsteamer) [inline dpgen/data/arginfo.py:136]: transforming
->
isotropic transforming

Comment (Vibsteamer) [inline dpgen/data/arginfo.py:136]: transforming
->
isotropic transforming

File: dpgen/data/arginfo.py [hunk 10]
```diff
@@ -35,6 +36,135 @@ def init_reaction_mdata_arginfo() -> Argument:
...
+    doc_skip_relax = "If it's true, you may directly run stage 2 (perturb and scale) using an unrelaxed POSCAR."
+    doc_pert_numb = "Number of perturbations for each POSCAR."
```
Comment (Vibsteamer) [inline dpgen/data/arginfo.py:138]: Number of perturbations for each POSCAR
->
Number of perturbations for each scaled (key scale) POSCAR

File: dpgen/data/arginfo.py [hunk 11]
```diff
@@ -35,6 +36,135 @@ def init_reaction_mdata_arginfo() -> Argument:
...
+    doc_pert_box = "Percentage of Perturbation for cells."
+    doc_pert_atom = "Perturbation of each atoms (Angstrom)."
```
Comment (Vibsteamer) [inline dpgen/data/arginfo.py:140]: Perturbation of atom coordinates (Angstrom). Random perturbations are performed on
three coordinates of each atom by adding values randomly sampled from a uniform distribution in the range [−pert_atom, pert_atom].

File: dpgen/data/arginfo.py [hunk 12]
```diff
@@ -35,6 +36,135 @@ def init_reaction_mdata_arginfo() -> Argument:
...
+    doc_vacuum_resol = "Interval of thickness of vacuum. If size of `vacuum_resol` is 1, the interval is fixed to its value. If size of `vacuum_resol` is 2, the interval is `vacuum_resol[0]` before `mid_point`, otherwise `vacuum_resol[1]` after `mid_point`."
+    doc_vacuum_numb = "The total number of vacuum layers **Necessary** if vacuum_resol is empty."
+    doc_mid_point = "The mid point of head region and tail region. **Necessary** if the size of vacuum_resol is 2 or 0."
+    doc_head_ratio  = "Ratio of vacuum layers in the nearby region with denser intervals(head region). **Necessary** if vacuum_resol is empty."
+    doc_millers = "Miller indices."
+    doc_relax_incar = "Path of INCAR for relaxation in VASP. **Necessary** if `stages` include 1."
+    doc_scale = "Scales for transforming cells."
+    doc_skip_relax = "If it's true, you may directly run stage 2 (perturb and scale) using an unrelaxed POSCAR."
+    doc_pert_numb = "Number of perturbations for each POSCAR."
+    doc_pert_box = "Percentage of Perturbation for cells."
```
Comment (Vibsteamer) [inline dpgen/data/arginfo.py:139]: Anisotropic Perturbation for cells (independent changes of lengths of three box vectors as well as angel among) in decimal formats. 9 elements of the 3x3 perturbation matrix will be randomly sampled from a uniform distribution (default) in the range [-pert_box, pert_box]. Such a perturbation matrix adds the identity matrix gives the actual transformation matrix for this perturbation operation.

File: dpgen/data/arginfo.py (hunk 9 above)
Comment (Vibsteamer) [inline dpgen/data/arginfo.py:136]: transforming
->
isotropic transforming

File: dpgen/data/arginfo.py (hunk 10 above)
Comment (Vibsteamer) [inline dpgen/data/arginfo.py:138]: Number of perturbations for each POSCAR
->
Number of perturbations for each scaled (key scale) POSCAR

File: dpgen/data/arginfo.py (hunk 12 above)
Comment (Vibsteamer) [inline dpgen/data/arginfo.py:139]: Anisotropic Perturbation for cells (independent changes of lengths of three box vectors as well as angel among) in decimal formats. 9 elements of the 3x3 perturbation matrix will be randomly sampled from a uniform distribution (default) in the range [-pert_box, pert_box]. Such a perturbation matrix adds the identity matrix gives the actual transformation matrix for this perturbation operation.

File: dpgen/data/arginfo.py (hunk 11 above)
Comment (Vibsteamer) [inline dpgen/data/arginfo.py:140]: Perturbation of atom coordinates (Angstrom). Random perturbations are performed on
three coordinates of each atom by adding values randomly sampled from a uniform distribution in the range [−pert_atom, pert_atom].

File: dpgen/data/surf.py [hunk 13]
```diff
@@ -477,13 +477,13 @@ def pert_scaled(jdata) :
           raise RuntimeError("the length of vacuum_resol must equal 1 or 2")
           
     else:         
-       vacuum_num = jdata['vacuum_numb']
-       head_ratio = jdata['head_ratio']
-       mid_point = jdata['mid_point']
+       vacuum_num = jdata['vacuum_numb'] # the total number of vacuum layers
+       head_ratio = jdata['head_ratio'] # the nearby region with denser intervals (head region)
```
Comment (Vibsteamer) [inline dpgen/data/surf.py:481]: \# deciding the mid_point by `vacum_max` * `head_ratio`, which poiny separates the nearby region with denser intervals (head region) and the far-away region with sparser intervals (tail region).

**Review by Vibsteamer — CHANGES_REQUESTED** [review]
best wishes to your endeavors

**Comment by HuangJiameng** [comment]
> best wishes to your endeavors

Thank you for your comprehensive review and suggestions!

**Review by Vibsteamer — APPROVED** [review]

**Review by AnguseZhang — APPROVED** [review]

**Pull request merged**